lib/i18n/js.rb

```diff
@@ -76,32 +85,26 @@ def self.filtered_translations
           Utils.deep_merge!(result, segment.translations)
         end
       end
-      return Utils.deep_key_sort(translations) if I18n::JS.sort_translation_keys?
+      return Utils.deep_key_sort(translations) if I18n::JS.configuration.sort_translation_keys?
       translations
     end
 
     def self.translation_segments
-      if config? && config[:translations]
-        configured_segments
-      else
-        [Segment.new("#{DEFAULT_EXPORT_DIR_PATH}/translations.js", translations)]
-      end
-    end
+      configuration.translations.inject([]) do |segments, options|
+        file = options[:file]
+        only = options[:only] || '*'
```

Comment:
I will do more refactor once this is merged
But I don't want to change too many things in one PR